### DISCUSSION TITLE
Ingredient variants shorthand

### DISCUSSION THREAD

**Discussion by wkokolis** [cooklang/spec#74] [Ideas]
Right now if we have recipes that call for specific types of ingredients, such as rolled oats vs. steel-cut oats, we have to list all of them in the aisle definition and write out the full names in the recipes.  While we can do something like `steel-cut #oats` in the recipe if we don't want to have to write out everything, that doesn't translate to the shopping list.

Both options are a little awkward and the more common practice in cookbooks to denote ingredients as "oats, steel-cut" or "oats (steel-cut)" only in the ingredient list.  In the cooking steps, you just say "oats".

I propose the ingredient syntax be updated to include variants/adjectives, perhaps separate by a pipe?

Something like this:

```
Combine @oats{6%cups|steel-cut}, @raisins{1%cup|golden}, and @coconut{2%cups|shredded} in a #bowl.
``` 

In the recipe steps it would render as:
```
Combine oats, raisins, and coconut in a bowl.
```  

In the shopping list, however, you'd see:
```
oats, steel-cut      6 cups
raisins, golden      1 cup
coconut, shredded      2 cups
```

And because the core ingredients are just "oats", "raisins", and "coconut", that's all you'd have to list in the aisle definition.

**Comment by dubadub** [comment]
Yeah, I think it could be useful. I'm linking related discussion https://github.com/cooklang/spec/discussions/57.

**Comment by dubadub** [comment]
Relevant proposal for spec change https://github.com/cooklang/spec/pull/106.

**Comment by dubadub** [comment]
This is now part of the main specification, closing the discussion.
